prompt.sh

```
@@ -1,32 +1,326 @@
+
+  if [[ "$local_branch" != "HEAD" ]]; then
+    local upstream_branch=$(git rev-parse --abbrev-ref "@{upstream}" 2> /dev/null)
+    # If we get back what we put in, then that means the upstream branch was not found.  (This was observed on git 1.7.10.4 on Ubuntu)
+    [[ "$upstream_branch" = "@{upstream}" ]] && upstream_branch=''
+    # If the branch is not tracking a specific remote branch, then assume we are tracking origin/[this_branch_name]
+    [[ -z "$upstream_branch" ]] && upstream_branch="origin/$local_branch"
+    if [[ -n "$upstream_branch" ]]; then
+      # Check if the upstream exists or not
+      if git rev-parse "$upstream_branch" > /dev/null 2>&1; then
+        # Upstream branch exists, so we will count how many commits from the current branch are not on the upstream
+        git_ahead_count=$(git rev-list --count "${base_branch_for_ahead_count}..${local_branch}" 2> /dev/null)
```

Shellcheck (0.11.0)

[warning] base_branch_for_ahead_count is referenced but not assigned.

(SC2154)